library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -402,10 +402,12 @@ private void downloadOrContinueBatch(List<FileDownloadInfo> downloads) {
     private void download(FileDownloadInfo info) {
         Uri downloadUri = ContentUris.withAppendedId(downloadsUriProvider.getAllDownloadsUri(), info.getId());
         FileDownloadInfo.ControlStatus.Reader controlReader = new FileDownloadInfo.ControlStatus.Reader(getContentResolver(), downloadUri);
+        DownloadBatch downloadBatch = batchRepository.retrieveBatchFor(info);
```

Comment:
So this is called every time the service is notified, but I think we cannot get around it (performance wise).